Core/Utilities/DEncode.py

```diff
@@ -16,236 +16,369 @@
 
 import types
 import datetime
+import os
+
+import inspect
+import traceback
+from pprint import pprint
+
+
+# Setting this environment variable will enable the dump of the debugging
+# call stack
+DIRAC_DEBUG_DENCODE_CALLSTACK = bool(os.environ.get('DIRAC_DEBUG_DENCODE_CALLSTACK', False))
+
+
+def printDebugCallstack():
+  """ Prints information about the current stack as well as the caller parameters.
+      The purpose of this method is to track down all the places in DIRAC that might
+      not survive the change to JSON encoding.
+
+      :returns: nothing, just prints
+
+  """
+  def stripArgs(frame):
+    """ Keeps only the parameters and their values from a frame
+
+        :param frame: frame object
+
+        :returns: dict {param name: value}
+    """
+    # Get all the arguments of the call
+    allArgs = inspect.getargvalues(frame)
+    # Keep only the arguments that are parameters of the call, as well as their value
+    return dict([(argName, allArgs.locals[argName]) for argName in allArgs.args])
+
+  tb = traceback.format_stack()
+  frames = inspect.stack(context=100)
```

Comment:
why 100, is that the maximum stack level? or just not expected to be exceeded?

Reply:
Hum, my bad, I should have put a name there. It's just that the default is 1, so you do not see far enough by default